packages/ra-core/src/form/getFormInitialValues.ts

```diff
@@ -10,6 +10,12 @@ export default function getFormInitialValues(
         getValues(defaultValues, record),
         record
     );
+    // replace null values by empty string to avoid controlled/ uncontrolled input warning
+    Object.keys(finalInitialValues).forEach(key => {
+        if (finalInitialValues[key] === null) {
+            finalInitialValues[key] = '';
+        }
+    });
```

Comment:
This doesn't work for nested values, like `author.name` for instance

Reply:
So the fix requires a bit of recursion: 

```jsx
export default function getFormInitialValues(
    defaultValues: DefaultValue,
    record?: Partial<RaRecord>
) {
    const finalInitialValues = merge(
        {},
        getValues(defaultValues, record),
        record
    );
    // replace null values by empty string to avoid controlled/ uncontrolled input warning
    return mapValuesDeep(finalInitialValues, value =>
        value === null ? '' : value
    );
}

const mapValuesDeep = (obj, callback) =>
    mapValues(obj, v =>
        isPlainObject(v) ? mapValuesDeep(v, callback) : callback(v)
    );
```

But I'm afraid it could have nasty side effects. Why should we modify the inner properties of embedded objects - some of which my not be modifiable? Besides, the `sanitizeEmptyValues` doesn't work for nested properties, so the empty string would be sent back to the server. 

All in all, this is another corner case, which already exists today, and I don't think this PR makes it worse.